hydra-cluster/src/CardanoClient.hs

```diff
@@ -111,25 +111,22 @@ waitForUTxO node utxo =
     txOut ->
       error $ "Unexpected TxOut " <> show txOut
 
-mkGenesisTx ::
+mkInitialTx ::
```

Comment:
This is NOT a good name. We have a protocol transaction that is exactly called this.

Also, the error call still calls this "genesis transaction".

Suggestion: `buildRaw` is only used in this function. Let's move it into this function and call this `buildRawTransaction` or so. Adding a haddock could also help to explain what the difference between this, `buildTransaction` and `mkGenesisTx` is (if we even keep the latter).